src/Aspire.Hosting/Dcp/DcpHostService.cs

```diff
@@ -197,23 +199,31 @@ private static Socket CreateLoggingSocket(string socketPath)
 
     private async Task StartLoggingSocketAsync(Socket socket)
     {
-        while (true)
+        await Task.CompletedTask.ConfigureAwait(ConfigureAwaitOptions.ForceYielding);
+        List<Task> outputLoggers = [];
+        while (!_shutdownCts.IsCancellationRequested)
         {
             try
             {
-                Socket acceptedSocket = await socket.AcceptAsync().ConfigureAwait(false);
-                _ = Task.Run(() => LogSocketOutputAsync(acceptedSocket), CancellationToken.None);
+                Socket acceptedSocket = await socket.AcceptAsync(_shutdownCts.Token).ConfigureAwait(false);
+                outputLoggers.Add(LogSocketOutputAsync(acceptedSocket, _shutdownCts.Token));
             }
             catch
             {
                 // Suppress exceptions reading logs from DCP controllers
             }
         }
+
+        await Task.WhenAll(outputLoggers).ConfigureAwait(false);
+        socket.Dispose();
     }
 
-    private async Task LogSocketOutputAsync(Socket socket)
+    private async Task LogSocketOutputAsync(Socket socket, CancellationToken cancellationToken)
     {
-        var reader = PipeReader.Create(new NetworkStream(socket));
+        await Task.CompletedTask.ConfigureAwait(ConfigureAwaitOptions.ForceYielding);
```

Comment:
Add a comment why this is here

Why is this better than calling method in a `Task.Run`?